prow/cmd/tide/main.go

```diff
@@ -105,7 +108,11 @@ func main() {
 	}
 	defer gc.Clean()
 
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, os.Interrupt, syscall.SIGTERM)
+
 	c := tide.NewController(ghcSync, ghcStatus, kc, configAgent, gc, logger)
+	defer c.Shutdown()
```

Comment:
I don't think this will work since you lack graceful shutdown for `ListenAndServe` (which is its own bag of worms).

https://golang.org/pkg/net/http/#Server.Shutdown